### PR TITLE
feat: ScaladocCompletion for Scala3

### PR DESCRIPTION
**Pull request by tanishiking**
close https://github.com/scalameta/metals/issues/3955

scaladoc

Overview, basically the same with Scala2 implementation:

- if the cursor follows `/**`, insert `*/` on the cursor and compile it using `InteractiveDriver`
  - We need to close the multiline comment, otherwise, the following code will be commented out, and we can't retrieve the tree of the code under `/**`.
- Find the class/method definition we're documenting
  - the class/method that defined just below the cursor
    - (I simplified the algorithm than Scala2 one. In Scala2, we try to find the class/method that is defined closest to the cursor)
- Extract the method or class constructor parameters, and construct the text.

### CODE REVIEW

File: mtags/src/main/scala-3/scala/meta/internal/pc/completions/ScaladocCompletions.scala [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+   * @return
+   */
+  def isScaladocCompletion(pos: SourcePosition, text: String): Boolean =
+    pos.point >= 3 &&
+      text.charAt(pos.point - 3) == '/' &&
+      text.charAt(pos.point - 2) == '*' &&
+      text.charAt(pos.point - 1) == '*'
+
+  class AssociatedMemberDefFinder(pos: SourcePosition) extends TreeTraverser:
+    private var associatedDef: Option[MemberDef] = None
```
Comment (tgodzik) [inline mtags/src/main/scala-3/scala/meta/internal/pc/completions/ScaladocCompletions.scala:88]: Does it need to be a var? Maybe we could return it from traverse?

**Review by tgodzik — COMMENTED** [review]
Looks already great!

File: mtags/src/main/scala-3.2/scala/meta/internal/pc/MetalsSignatures.scala [hunk 2]
```diff
@@ -1,14 +1,15 @@
...
-import dotty.tools.dotc.util.SourcePosition
 import dotty.tools.dotc.core.Contexts.*
+import dotty.tools.dotc.core.Denotations.*
+import dotty.tools.dotc.core.Flags
 import dotty.tools.dotc.util.Signatures
 import dotty.tools.dotc.util.Signatures.Signature
-import scala.meta.internal.mtags.MtagsEnrichments.*
-import scala.meta.pc.SymbolSearch
-import dotty.tools.dotc.core.Flags
-import dotty.tools.dotc.core.Denotations.*
```
Comment (tanishiking) [inline mtags/src/main/scala-3.2/scala/meta/internal/pc/MetalsSignatures.scala:11]: Fixed by `scalafixAll`

File: mtags/src/main/scala-3/scala/meta/internal/pc/completions/CompletionValue.scala [hunk 3]
```diff
@@ -23,6 +23,7 @@ sealed trait CompletionValue:
   final def completionItemKind(using Context): CompletionItemKind =
     this match
       case _: CompletionValue.Keyword => CompletionItemKind.Keyword
+      case _: CompletionValue.Document => CompletionItemKind.Snippet
```
Comment (tanishiking) [inline mtags/src/main/scala-3/scala/meta/internal/pc/completions/CompletionValue.scala:26]: Not sure which `CompletionItemKind` is the most appropriate for doc string 🤔

**Comment by tanishiking** [comment]
Let's hold off until 0.11.7 is released

**Pull request merged**